### PR TITLE
xen-cbmc: fix ci script

### PR DESCRIPTION
**Pull request by nmanthey**
This script has been broken, and not been picked up in CI until now.

This change fixes the script, as well as makes sure that the relevant
steps are executed, and the resulting build artefacts have the required
goto-cc section in them.

This CI script has been refered to as part of the paper:
"Using model checking tools to triage the severity of security bugs in
the Xen hypervisor", that has been presented at FMCAD 2020.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by kroening — APPROVED** [review]

**Review by NlightNFotis — CHANGES_REQUESTED** [review]
Thanks for the fix! I would say it's much needed 👍 

May I ask why the file was renamed from `.yml` to `.yaml`? This has the unfortunate side effect of breaking file revision history, which makes it hard to detect what the actual change here is (the one that actually fixed the broken CI issue).

File: .github/workflows/build-and-test-Xen.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        run: git clone git://xenbits.xen.org/xen.git xen_4_13 && cd xen_4_13 && git reset --hard RELEASE-4.13.0 && pwd
+
+      - name: Prepare compile Xen with CBMC via one-line-scan
+        run: |
+          ln -s goto-cc src/goto-cc/goto-ld
+          ln -s goto-cc src/goto-cc/goto-as
+          ln -s goto-cc src/goto-cc/goto-g++
+
+      - name: Compile Xen with CBMC via one-line-scan, and check for goto-cc section
+        run: one-line-scan/one-line-scan --add-to-path $(pwd)/src/cbmc --add-to-path $(pwd)/src/goto-diff --add-to-path $(pwd)/src/goto-cc --no-analysis --trunc-existing --extra-cflags -Wno-error -o CPROVER -j 3 -- make -C xen_4_13 xen -j $(nproc) -k || true
```
Comment (hannes-steffenhagen-diffblue) [inline .github/workflows/build-and-test-Xen.yaml]: Could you split this over multiple lines? This is a bit hard to read.

**Review by martin-cs — APPROVED** [review]
Does this mean I will stop getting e-mails saying "Run failed: Build and Test Xen" whenever I push to a PR?

File: .github/workflows/build-and-test-Xen.yaml (hunk 1 above)
Comment (nmanthey) [reply .github/workflows/build-and-test-Xen.yaml]: Yes, will split.

**Comment by nmanthey** [comment]
> Thanks for the fix! I would say it's much needed +1
> 
> May I ask why the file was renamed from `.yml` to `.yaml`? This has the unfortunate side effect of breaking file revision history, which makes it hard to detect what the actual change here is (the one that actually fixed the broken CI issue).

I will split the commit into (1) file rename, and (2) actual diff. The rename is basically for consistency.

Yes, I can split that over multiple lines.

Yes, this fix should stop those PR emails (I hope, I did not push PRs recently ;) )

**Review by NlightNFotis — APPROVED** [review]
Thanks for the explanation. 

Good to go from me 👍

**Pull request merged**